src/plugins/intel_cpu/src/graph_dumper.cpp

```diff
@@ -358,6 +358,11 @@ void average_counters(const Graph &graph) {
      */
     static int graphIndex = 0;
 
+    if (!graph.getGraphContext()) {
+        DEBUG_LOG("graph.m_context is null. Don't dump average_counters.");
+        return;
+    }
+
```

Comment:
hi @maxnick ,` void average_counters(const Graph &graph)` is called here(https://github.com/xipingyan/openvino/blob/2b1cfb5fb273f3a4571213484718316050930c52/src/plugins/intel_cpu/src/graph.cpp#L61).
ReadValueWithSubgraph has subgraph node, it's context is from here(https://github.com/xipingyan/openvino/blob/2b1cfb5fb273f3a4571213484718316050930c52/src/plugins/intel_cpu/src/nodes/memory.cpp#L679).

When https://github.com/xipingyan/openvino/blob/2b1cfb5fb273f3a4571213484718316050930c52/src/plugins/intel_cpu/src/graph.cpp#L61 is called, the ReadValueWithSubgraph's context has been release in main graph, so the context is null  here.

Reply:
> When https://github.com/xipingyan/openvino/blob/2b1cfb5fb273f3a4571213484718316050930c52/src/plugins/intel_cpu/src/graph.cpp#L61 is called, the ReadValueWithSubgraph's context has been release in main graph, so the context is null here.

The context is stored as a `shared_ptr`, so if it's somehow released within the destructor of the main graph, the context still exists till the reference counter is zero. So it's still not clear how it can be null except a situation when it has never been initialized.

Reply:
I am also very strange, shared_ptr can hold the ptr. I have to dig it up again.
I found real problem is here: https://github.com/xipingyan/openvino/blob/ef5962b2360f4b3283087dc4046eed4108846dbb/src/plugins/intel_cpu/src/graph_optimizer.cpp#L3296
`MemoryInput` is replaced with `MemoryInputSingle` in the funciton `DropRedundantMemoryOutput`, and  `MemoryInput` will be released before` MemoryInput::initOptimalPrimitiveDescriptor()` is not called, so the context is empty in subgraph's deconstructor.